pkg/controller/policyrecommendation/policyrecommendation_controller.go

```diff
@@ -0,0 +1,464 @@
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceValidationError, fmt.Sprintf("failed to retrieve / validate  %s", render.ManagerInternalTLSSecretName), err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	esgwCertificate, err := certificateManager.GetCertificate(r.client, relasticsearch.PublicCertSecret, common.OperatorNamespace())
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceValidationError, fmt.Sprintf("Failed to retrieve / validate  %s", relasticsearch.PublicCertSecret), err, reqLogger)
+		return reconcile.Result{}, err
+	} else if esgwCertificate == nil {
+		log.Info("Elasticsearch gateway certificate is not available yet, waiting until they become available")
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Elasticsearch gateway certificate are not available yet, waiting until they become available", nil, reqLogger)
+		return reconcile.Result{}, nil
+	}
+	trustedBundle := certificateManager.CreateTrustedBundle(managerInternalTLSSecret, esgwCertificate)
+
+	var policyRecommendationServerCertSecret certificatemanagement.KeyPairInterface
+	if managementClusterConnection == nil {
+		policyRecommendationServerCertSecret, err = certificateManager.GetOrCreateKeyPair(
+			r.client,
+			render.PolicyRecommendationCertSecret,
+			common.OperatorNamespace(),
+			dns.GetServiceDNSNames(render.PolicyRecommendationName, render.PolicyRecommendationNamespace, r.clusterDomain))
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceValidationError, fmt.Sprintf("failed to retrieve / validate  %s", render.PolicyRecommendationCertSecret), err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+	certificateManager.AddToStatusManager(r.status, render.PolicyRecommendationNamespace)
+
+	// Fetch the Authentication spec. If present, we use to configure user authentication.
+	authenticationCR, err := utils.GetAuthentication(ctx, r.client)
+	if err != nil && !errors.IsNotFound(err) {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying Authentication", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+	if authenticationCR != nil && authenticationCR.Status.State != operatorv1.TigeraStatusReady {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, fmt.Sprintf("Authentication is not ready - authenticationCR status: %s", authenticationCR.Status.State), nil, reqLogger)
+		return reconcile.Result{}, nil
+	}
+
+	// Create a component handler to manage the rendered component.
+	handler := utils.NewComponentHandler(log, r.client, r.scheme, instance)
+
+	keyValidatorConfig, err := utils.GetKeyValidatorConfig(ctx, r.client, authenticationCR, r.clusterDomain)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceValidationError, "Failed to process the authentication CR.", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	reqLogger.V(3).Info("rendering components")
+	hasNoLicense := !utils.IsFeatureActive(license, common.PolicyRecommendationFeature)
```

Comment:
Does our apiserver have support for PolicyRecommendationFeature?

Reply:
https://github.com/tigera/calico-private/tree/epic/continuous-policy-recommendation/apiserver/pkg/registry/projectcalico/policyrecommendationscope